### PR TITLE
feat(server): Projects Recycle bin

### PR DESCRIPTION
**Pull request by hexaforce**
# Overview
Add `deleted` flag to project

## What I've done
1. Deletion flag added to Project.

2. To update the delete flag, please use the `UpdateProject` mutation. 
Please note that the existing deleteProject mutation physically deletes the project from the database, so caution is required.

3. I have added `DeletedProjects` to the GraphQL query. 
This retrieves all entries where the delete flag is set to true.

4. I have removed the `includeArchived` flag from the GetProjects GraphQL query. 
While the frontend currently does not make queries using includeArchived, the backend will no longer support it.

## How I tested
Testing with Test Code

## Which point I want you to review particularly

Please remove `includeArchived` from the web code.
The update for the delete flag will be similar to the usage of `ARCHIVE_PROJECT`.
`DeletedProjects` will be used similarly to `StarredProjects`.

## Memo
https://www.notion.so/eukarya/BE-Projects-Recycle-bin-82ca047f781740febd354db58aa2acea

## Release Notes

- **New Features**
  - Introduced functionality to manage project deletion status, including the ability to mark projects as deleted and retrieve deleted projects.
  - Added a new query for fetching deleted projects by team ID.
  - Enhanced project management capabilities with updated GraphQL schema and API.
  - New focused testing approach with the addition of a `failcheck` target for unit tests.

- **Bug Fixes**
  - Improved error handling and transaction management during project deletion.

- **Tests**
  - Expanded test coverage for project creation and import functionalities to include new deletion attributes.

These updates enhance user experience by providing better project management tools and ensuring data integrity.

### CODE REVIEW

File: server/e2e/gql_project_test.go [hunk 1]
```diff
@@ -452,3 +452,83 @@ func TestSortByUpdatedAt(t *testing.T) {
 	edges.Element(1).Object().Value("node").Object().Value("name").Equal("project3-test")
 	edges.Element(2).Object().Value("node").Object().Value("name").Equal("project1-test")
 }
+
+//  go test -v -run TestDeleteProjects ./e2e/...
```
Comment (pyshx) [inline server/e2e/gql_project_test.go:456]: no need of this

File: server/gql/project.graphql [hunk 2]
```diff
@@ -140,13 +142,13 @@ type ProjectEdge {
 extend type Query {
   projects(
     teamId: ID!
-    includeArchived: Boolean
     pagination: Pagination
     keyword: String
     sort: ProjectSort
-  ): ProjectConnection!
+  ): ProjectConnection! # not included deleted projects
```
Comment (pyshx) [inline server/gql/project.graphql:148]: we don't need this comment

File: server/go.mod [hunk 3]
```diff
@@ -23,6 +23,7 @@ require (
 	github.com/kennygrant/sanitize v1.2.4
 	github.com/labstack/echo/v4 v4.11.4
 	github.com/mitchellh/mapstructure v1.5.0
+	github.com/oklog/ulid v1.3.1
```
Comment (pyshx) [inline server/go.mod:26]: why is this direct dependency now?

**Review by pyshx — APPROVED** [review]
Thanks, the implementation looks good to me.
Please fix the comments.

File: server/e2e/gql_project_test.go (hunk 1 above)
Comment (hexaforce) [reply server/e2e/gql_project_test.go:456]: yes

File: server/Makefile [hunk 4]
```diff
@@ -20,6 +20,7 @@ lint:
 test:
 	go test -race -short -v ./...
 
+# make failcheck TEST_DIR=./internal/usecase/interactor
```
[the remark this answers is not in the document]
Comment (hexaforce) [reply server/Makefile:23]: yes

File: server/gql/project.graphql (hunk 2 above)
Comment (hexaforce) [reply server/gql/project.graphql:148]: yes

File: server/go.mod (hunk 3 above)
Comment (hexaforce) [reply server/go.mod:26]: Okay, I'll put it back.
For some reason, in my environment, it happens when I run `make gql`.

**Review by mkumbobeaty — APPROVED** [review]

**Pull request merged**